components/screens/deliveries/EditDelivery.tsx

```diff
@@ -136,12 +138,22 @@ class EditDelivery extends React.Component<Props, State> {
     this.setState({ loading: true });
     const Api = new PakkasmarjaApi();
     const productsService = Api.getProductsService(accessToken.access_token);
-    const unfilteredProducts: Product[] = await productsService.listProducts(undefined, itemGroupCategory, accessToken.userId, undefined, 999);
-    const products = unfilteredProducts.filter(product => product.active === true);
-    const deliveryPlacesService = Api.getDeliveryPlacesService(accessToken.access_token);
-    const deliveryPlaces = await deliveryPlacesService.listDeliveryPlaces();
     const productPricesService = Api.getProductPricesService(accessToken.access_token);
-    const productPrice: ProductPrice[] = products[0] ? await productPricesService.listProductPrices(products[0].id || "", "CREATED_AT_DESC", undefined, undefined, 1) : [];
+    const deliveryPlacesService = Api.getDeliveryPlacesService(accessToken.access_token);
+    const deliveryQualitiesService = Api.getDeliveryQualitiesService(accessToken.access_token);
+    
+    const [ unfilteredProducts, deliveryPlaces ] = await Promise.all([
+      productsService.listProducts(undefined, itemGroupCategory, accessToken.userId, undefined, 999),
+      deliveryPlacesService.listDeliveryPlaces()
+    ]);
+
+    const products = unfilteredProducts.filter(product => product.active === true);
+
+    const [ productPrice, deliveryQualities ] = await Promise.all([
+      products[0] ? await productPricesService.listProductPrices(products[0].id || "", "CREATED_AT_DESC", undefined, undefined, 1) : [],
+      (await deliveryQualitiesService.listDeliveryQualities(ItemGroupCategory.FRESH, products[0].id || "")).reverse()
```

Comment:
sort instead by descending priceBonus amount

Reply:
Reversing array removed from this line